src/sawtooth/sawtooth.js

```diff
@@ -14,6 +14,152 @@ const log = require('../comm/util.js').log;
+ * @param {Object} msg event message
+ * @return {void}
+ */
+async function handleEvent(msg) {
+    if (msg.messageType === Message.MessageType.CLIENT_EVENTS) {
+        const events = EventList.decode(msg.content).events;
+        getBlock(events).then(result => {
+            lastKnownBlockId = result.blockId.toString();
+            let blockNum=result.blockNum;
+            //On receiving event with block, update the status of the block to sucess
+            blockCommitSatus.set(blockNum, 'success');
+        });
+    } else {
+        log('Warn: Received message of unknown type:', msg.messageType);
+    }
+}
+
+/**
+ * Subscribe to block-commit delta events
+ * @param {Object} stream object to send event subscribe message
+ * @return {void}
+ */
+async function subscribe(stream) {
+    //Subscribe to block-commit delta event
+    const blockSub = EventSubscription.create({
+        eventType: 'sawtooth/block-commit'
+    });
+
+    if(lastKnownBlockId === null) {
+        await getCurrentBlockId().then(() => {
+        });
+    }
+    stream.send(
+        Message.MessageType.CLIENT_EVENTS_SUBSCRIBE_REQUEST,
+        ClientEventsSubscribeRequest.encode({
+            subscriptions: [blockSub],
+            lastKnownBlockIds: [lastKnownBlockId]
+        }).finish()
+    )
+        .then(response => ClientEventsSubscribeResponse.decode(response))
+        .then(decoded => {
+            const status = _.findKey(ClientEventsSubscribeResponse.Status,
+                val => val === decoded.status);
+            if (status !== 'OK') {
+                throw new Error(`Validator responded with status "${status}"`);
+            }
+        });
+}
+
+/**
+ * Unsubscribe to block-commit delta events
+ * @param {Object} stream1 object to send event unsubscribe message
+ * @return {void}
+ */
+function unsubscribe(stream1) {
+    stream1.send(
+        Message.MessageType.CLIENT_EVENTS_UNSUBSCRIBE_REQUEST,
+        ClientEventsUnsubscribeRequest.encode({
+        }).finish()
+    )
+        .then(response => ClientEventsUnsubscribeResponse.decode(response))
+        .then(decoded => {
+            const status = _.findKey(ClientEventsUnsubscribeResponse.Status,
+                val => val === decoded.status);
+            if (status !== 'OK') {
+                throw new Error(`Validator responded with status "${status}"`);
+            }
+        });
+}
+/**
+ * Get batch commit event message on block commit
+ * @param {Number} block_num of next block
+ * @param {Number} batchStats Batch status object to update commit status
+ * @return {Promise<object>} returns batch commit status
+ */
+function getBatchEventResponse(block_num, batchStats) {
+    return new Promise(resolve => {
+        while(blockCommitSatus.get(block_num) !== 'pending') {
+        }
+        //remove the block number from map because we are done with this block
+        blockCommitSatus.delete(block_num);
+        batchStats.SetStatusSuccess();
```

Comment:
Why the next block is not pending, the batch commit status is success? Why the next block must include the batch?

Reply:
Once batch is submitted then the adding status of batch as pending to hash map and when we get the event then remove the entry from hash map and update the status to success and send it to caliper.

Reply:
Yes ,I know the total process. But Which event? I can't find the batch commit event, but the block commit event. So my question is why we get the  block commit event, we could update the status of the batch to success.

Reply:
We are subscribed to block-commit event. In sawtooth it will be block level, not batch level. Multiple batches are added to block and it is committed. After getting block commit event we are setting batch status to success and sending it to caliper.

Reply:
Got your point, i am updating batch status to pending when submitted but if multiple batches goes to in single block then we are not updating the status of all batches, that is  question right?

Reply:
Yes that's correct. It should be fine because block consists of batches and each batch contain multiple transactions and block is ultimate entity getting added to block chain.

Reply:
As the codes, when we get block(2) commit event, we are setting all of batches(a, b, c, d, e, f) status to success, but I think this is not correct.  The truth is that only (a, b, c) batches status is success. But the codes are setting all of the batches' status to success.